src/Transactions/Deserializer.php

```diff
@@ -47,19 +54,35 @@ public static function new(string $serialized)
      */
     public function deserialize(): AbstractTransaction
     {
-        $data = [];
+        $decodedRlp = RlpDecoder::decode($this->encodedRlp);
 
-        $this->deserializeCommon($data);
+        $data = [];
 
-        $this->deserializeData($data);
+        $data['network']          = $this->parseNumber($decodedRlp[0]);
+        $data['nonce']            = $this->parseBigNumber($decodedRlp[1]);
+        $data['gasPrice']         = $this->parseNumber($decodedRlp[3]);
+        $data['gasLimit']         = $this->parseNumber($decodedRlp[4]);
+        $data['recipientAddress'] = $this->parseAddress($decodedRlp[5]);
+        $data['value']            = $this->parseBigNumber($decodedRlp[6]);
+        $data['data']             = $this->parseHex($decodedRlp[7]);
+
+        if (count($decodedRlp) === 12) {
+            $data['v'] = $this->parseNumber($decodedRlp[9]) + 27;
+            $data['r'] = $this->parseHex($decodedRlp[10]);
+            $data['s'] = $this->parseHex($decodedRlp[11]);
+        }
 
         $transaction = $this->guessTransactionFromData($data);
 
-        $this->deserializeSignatures($transaction->data);
+        $eip1559Prefix = '02'; // marker for Type 2 (EIP1559) transaction which is the standard nowadays
```

Comment:
could probably throw this into a constant up top `const EIP_1559_PREFIX = '02';`